frontend/pages/sites/SiteListItem.test.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { MemoryRouter } from 'react-router-dom';
+
+import SiteListItem from './SiteListItem';
+// import GitHubLink from '@shared/GitHubLink';
+// import PublishedState from './PublishedState';
```

Comment:
Should I be stubbing these instead?

Reply:
I think we can mock these to expect the props to be as expected. Ie

```jsx
import PublishedState from './PublishedState';

// Mock the component function
jest.mock('PublishedState', () => {
  return jest.fn(() => null)
})

// And then in the test
it('should render....', () => {
  expect(PublishedState).toHaveBeenCalledWith({ publishedAt: mockSite.publishedAt })
})
```

Reply:
I had to do this slightly differently with [expect.objectContaining()](https://jestjs.io/docs/expect#expectobjectcontainingobject) and [expect.anything()](https://dev.to/peterlidee/mocking-react-components-jest-mocking-react-part-2-2l8j#:~:text=The%20matcher%20.toHaveBeenCalledWith()%20receives%20two%20things%20from%20the%20mock%3A%20the%20props%20object%20and%20the%20ref%20value.%20We%20now%20pass%20into%20the%20matcher%20what%20we%20expect%20to%20find%3A%20an%20object%20with%20properties%20(to%20match%20the%20props)%20and%20an%20empty%20object%20%C2%B4%7B%7D%C2%B4%20to%20match%20the%20ref)

LMK what you think